src/python/test/test_extract_expression.py

```diff
@@ -78,8 +116,400 @@ def standard_positive_filter_test(self, json_name):
+        expected_error_message = ""
+        no_molec_exp_assay = self.proj_id + ":/Extract_Expression/no_molec_exp_assay"
+        command = [
+            "dx",
+            "extract_assay",
+            "expression",
+            no_molec_exp_assay,
+            "--list_assays",
+        ]
+
+        process = subprocess.Popen(
+            command, stderr=subprocess.PIPE, universal_newlines=True
+        )
+        actual_err_msg = process.communicate()[1]
+        # print(actual_err_msg)
+
+        self.assertTrue(expected_error_message in actual_err_msg)
+
+    # EM-14
+    # When file already exist
+    @unittest.skip("not yet implemented")
+    def test_output_already_exist(self):
+        # TODO replace this with a tempfile
+        output_path = os.path.join(
+            self.general_output_dir, "already_existing_output_file.tsv"
+        )
+        expected_error_message = (
+            "{} already exists. Please specify a new file path".format(output_path)
+        )
+        command = [
+            "dx",
+            "extract_assay",
+            "expression",
+            self.test_record,
+            "--output",
+            output_path,
+        ]
+
+        process = subprocess.Popen(
+            command, stderr=subprocess.PIPE, universal_newlines=True
+        )
+        actual_err_msg = process.communicate()[1]
+        # print(actual_err_msg)
+
+        self.assertTrue(expected_error_message in actual_err_msg)
+
+    # EM-15
+    # When a --retrieve-expression flag is passed without any value or when an empty JSON (an empty file or just {}) is passed with --retrieve-expression flag
+    # Note: empty JSON is tested in the JSON validation section
+    def test_no_value_retrieve_exp(self):
+        expected_error_message = "No filter json is passed with --retrieve-expression or JSON or --retrieve-expression does not contain valid filter information."
+        input_dict = {"path": self.test_record, "retrieve_expression": True}
+        self.standard_input_args_test(input_dict, expected_error_message)
+
+    # EM-16
+    # When the string provided is a malformed JSON
+    def test_malformed_retr_exp_json(self):
+        expected_error_message = (
+            "JSON provided for --retrieve-expression is malformatted."
+        )
+        command = [
+            "dx",
+            "extract_assay",
+            "expression",
+            self.test_record,
+            "--retrieve-expression",
+            r"{thisisbadjson",
+        ]
+
+        process = subprocess.Popen(
+            command, stderr=subprocess.PIPE, universal_newlines=True
+        )
+        actual_err_msg = process.communicate()[1]
+        # print(actual_err_msg)
+
+        self.assertTrue(expected_error_message in actual_err_msg)
+
+    # EM-17
+    # When the .json file provided does not exist
+    # Note: this probably needs to be tested with a Popen rather than with the ValidateArgsBySchema function
```

Comment:
Agreed! This should be probably just an E2E test and implemented once the main flow is finished.

Reply:
added @unittest.skip, we can reactivate this test in the future once the flow logic is in place